### ISSUE TITLE
Extension bug: getPopup gets ANY popup, can close wrong popups

### ISSUE THREAD

**Issue by danfinlay** [MetaMask/metamask-extension#4553]
On [this line](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/notification-manager.js#L96)

We should ensure that the popup we return is a MetaMask popup. This will ensure the other functions behave correctly.

**Comment by bdresser** [comment]
@danfinlay related to #4135 ?

**Comment by HangYang** [comment]
yeah, this is why the #4135 occurs.

**Comment by vittominacori** [comment]
Is anyone working on or are we waiting for PRs?

**Comment by danfinlay** [comment]
It does not appear anyone is working on this (no assignee, no comments announcing intention to work on this). We may post a bounty on this soon, but we also might solve it ourselves soon.  If you'd like to take a pass, you're welcome to.

**Comment by vittominacori** [comment]
It breaks my DApp logic so I could try to work on next week, having time...

Bounties are welcome :)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 0.15 ETH (75.27 USD @ $501.79/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MetaMask/metamask-extension/4553/642).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $25,905.82 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by bdresser** [comment]
a full solution should also address #4135, allowing the MetaMask popup to open even if another site's popup is already showing.

**Comment by bdresser** [comment]
full solution should also include a test for this behavior! 😺

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 12 months from now. Please review their questions below:


1. [shad-k](https://gitcoin.co/profile/shad-k) has started work. 	
 * Q: We can store a reference to the popup window when it is opened and then use that reference. I would like to know the exact steps to recreate this bug.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 0.15 ETH (75.14 USD @ $500.91/ETH) has been submitted by__:

 1. [@shad-k](https://gitcoin.co/profile/shad-k)



@bdresser please take a look at the submitted work:
* [PR](https://github.com/MetaMask/metamask-extension/pull/4597) by @shad-k

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MetaMask/metamask-extension/4553/642)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
$25,320.16 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by bdresser** [comment]
hey @shad-k, we've accepted @vittominacori's solution and merged #4592. but since we had a bit of a race condition with you claiming the bounty and both of you opening PRs, happy to pay out the sum to each of you.

@vittominacori perhaps you didn't see the post, but please claim the bounty on gitcoin if you're able! it helps us keep track of what's being worked on. feel free to send me an address 👍

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__The funding of 0.15 ETH (79.99 USD @ $533.29/ETH) attached to this issue has been approved & issued to @shad-k.__  

 * Learn more at [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MetaMask/metamask-extension/4553/642)
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $27,292.10 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by shad-k** [comment]
Hi @bdresser, really sorry for the confusion.
I had gone through @vittominacori 's solution, after I had started work, which seemed a little wrong and since he hadn't claimed the bounty I continued my work on it as I wasn't sure if he was still working on it. Seems even he has changed his solution since then and both of our implementations are pretty similar now. Thanks for paying out the bounty despite the confusion.
